### PR TITLE
Feature/lazyload react

### PR DESCRIPTION
**Pull request by Silind**
**What does this pull request introduce? Please describe**  
Lazyloading of React and ReactDOM modules in production.

As a part of the production build, React and ReactDOM will be extracted from the bundle, reducing the initial bundle size of the sample component to only **49KB**

React and ReactDOM will be lazyloaded from CDN, and the Direflow Component will render once the React and ReactDOM modules are ready.
This means, that the direflowBundle will no longer be render-blocking in any way when included in a host application.

Additionally, if React and ReactDOM are already included in the host application, Direflow will simply "hook into" these modules instead, and skip loading altogether.
This introduces a **major** performance improvement.

Running Direflow in development mode will remain as is; React and ReactDOM will still be included from node_modules.

**How did you verify that your changes work as expected? Please describe**  
I verified that the Direflow Component works in both development and production mode.

**Example**  
Please describe how we can try out your changes  
1. Create a new Direflow Setup.
2. Start the dev-server and verify that the Direflow Component renders.
3. Build the direflowBundle.
4. See that the /build/direflowBundle.js file is not exceeding 50KB.
5. Include the /build/direflowBundle.js file in another index.html file somewhere, and add the custom element to the body.
6. Open this index.html file in the browser and verify that the Direflow Component renders.
7. Go and inspect the DOM - see that the React and ReactDOM bundles are included in the header as an external script.

**Version**  
3.3.0

### CODE REVIEW

File: packages/direflow-component/src/services/asyncScriptLoader.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+type TBundle = { script: Element; hasLoaded: boolean };
+declare global {
+  interface Window {
+    wcPolyfillsLoaded: TBundle[];
+    reactBundleLoaded: TBundle[];
+  }
+}
+
+const asyncScriptLoader = (src: string, bundleList: TBundle[] | undefined): Promise<void> => {
```
Comment (JohnDeved) [inline packages/direflow-component/src/services/asyncScriptLoader.ts]: why not make this `bundleList: TBundle[] | undefined`

File: packages/direflow-component/src/config/config-overrides.ts [hunk 2]
```diff
@@ -10,24 +11,35 @@ export = function override(config: TConfig, env: string, options?: IOptions) {
...
+
   if (env === 'production') {
-    return config;
+    const [pathIndex] = config.entry;
+    const entryLoaderFile = fs.readFileSync(resolve(__dirname, './dist/config/entryLoader.js'), 'utf8');
+    const entryLoaderTemplate = handlebars.compile(entryLoaderFile);
+    const entryLoaderFileNew = entryLoaderTemplate({ pathIndex });
+    fs.writeFileSync(resolve(__dirname, './dist/config/entryLoader.js'), entryLoaderFileNew);
+
+    entry = [resolve(__dirname, './dist/config/entryLoader.js')];
```
Comment (JohnDeved) [inline packages/direflow-component/src/config/config-overrides.ts]: im having concerns about this. seems like we could maybe come up with a cleaner solution :)

File: packages/direflow-component/src/services/asyncScriptLoader.ts (hunk 1 above)
Comment (Silind) [reply packages/direflow-component/src/services/asyncScriptLoader.ts]: I think this is about communication.
`bundleList?:` would imply that the argument is optional.
That is not the case - but it’s gonna be a reference to a list on the global object that may not have been initialized yet, hence carries the _value_ of undefined.

If you look down a little further, you will see that it's getting initialized if not already set:
```javascript
if (!bundleList) {
  bundleList = [];
}
```

File: packages/direflow-component/src/config/config-overrides.ts (hunk 2 above)
Comment (Silind) [reply packages/direflow-component/src/config/config-overrides.ts]: Remember that this part of the code is executed in a Node process as a part of the Webpack build.
So this is not going to be run in the browser.
Can you elaborate on which parts that concerns you?

File: packages/direflow-component/src/config/config-overrides.ts [hunk 3]
```diff
@@ -113,6 +125,18 @@ function overrideResolve(currentResolve: IResolve) {
   return currentResolve;
 }
 
+function overrideExternals(externals: any, env: string) {
```
Comment (JohnDeved) [inline packages/direflow-component/src/config/config-overrides.ts]:
```suggestion
function overrideExternals(externals: { [key: string]: any }, env: string) {
```

**Pull request merged**